src/parser/jobs/sch/modules/DoTs.js

```diff
@@ -28,12 +29,140 @@ export default class DoTs extends CoreDoTs {
 		...DoTs.dependencies,
 		'checklist',
 		'suggestions',
+		'downtime',
 	]
 
 	static statusesToTrack = [
 		STATUSES.BIOLYSIS.id,
 	]
 
+	_lastBioCast = undefined
+	_application = {}
+
+	constructor(...args) {
+		super(...args)
+
+		const castFilter = {
+			by: 'player',
+			abilityId: ACTIONS.BIOLYSIS.id,
+		}
+		this.addEventHook('cast', castFilter, this._onTrackedDotCast)
+		const statusFilter = {
+			by: 'player',
+			abilityId: [STATUSES.BIOLYSIS.id],
+		}
+		this.addEventHook(['applydebuff', 'refreshdebuff'], statusFilter, this._onTrackedDotApply)
+	}
+
+	_pushApplication(targetKey, statusId, event) {
+		const target = this._application[targetKey] = this._application[targetKey] || {[STATUSES.BIOLYSIS.id]: []}
+		const source = this._lastBioCast
+		target[statusId].push({event, source})
+	}
+
+	_onTrackedDotCast(event) {
+		this._lastBioCast = event.ability.guid
+	}
+
+	_onTrackedDotApply(event) {
+		const statusId = event.ability.guid
+
+		// Make sure we're tracking for this target
+		const applicationKey = `${event.targetID}|${event.targetInstance}`
+		//save the application for later use in the output
+		this._pushApplication(applicationKey, statusId, event)
+	}
+
+	_formatDriftText(timestamp, previous=0) {
+		let drift = 0
+		const lateAllow = 1000
```

Comment:
Our convention is to define constants like this in the header (outside the module).  Typically something like `const LATE_ALLOW` or similar (personally, I'd be a little more verbose in the naming, but I won't block on exact names).

I'm also a little leery of the allowed ranges here - 1 second late reapplication doesn't allow for much wiggle room in terms of the time to finish your previous GCD as the DOT is expiring.  Our usual convention for re-use of oGCDs is to allow for up to 1 GCD worth of "drift" per cast, and I believe other DOT modules "forgive" up to 3 seconds of early re-application as convention.